src/App.js

```diff
@@ -37,77 +128,429 @@ function App(props) {
         }
     }
 
-    // Counter used for unique list item ids
-    // Increment every time a list item is added
-    const [counter, setCounter] = useState(500);
+    // Hidden modals for adding/editing/deleting list items
+    const [showAddItemModal, setShowAddItemModal] = useState(false);
+    const [showDeleteItemModal, setShowDeleteItemModal] = useState(false);
+
+    // Hidden modals for adding/editing/deleting lists
+    const [showAddListModal, setShowAddListModal] = useState(false);
+    const [showEditListModal, setShowEditListModal] = useState(false);
+    const [showDeleteListModal, setShowDeleteListModal] = useState(false);
 
-    const [showModal, setShowModal] = useState(false);
+    // Add item
+    const onAddItem = (text, priority) => {
+        setUndoStack(undoStack.concat([{
```

Comment:
You do this setting of undo stack often. Seems like a useful routine would be pushUndo({...})